plugin/pkg/shoot/mutator/admission.go

```diff
@@ -157,3 +227,28 @@ func addDeploymentTasks(shoot *core.Shoot, tasks ...string) {
 	}
 	controllerutils.AddTasks(shoot.Annotations, tasks...)
 }
+
+func (c *mutationContext) defaultShootNetworks(workerless bool) {
+	if c.seed != nil {
+		if c.shoot.Spec.Networking.Pods == nil && !workerless {
+			if c.seed.Spec.Networks.ShootDefaults != nil && c.seed.Spec.Networks.ShootDefaults.Pods != nil {
+				if cidrMatchesIPFamily(*c.seed.Spec.Networks.ShootDefaults.Pods, c.shoot.Spec.Networking.IPFamilies) {
+					c.shoot.Spec.Networking.Pods = c.seed.Spec.Networks.ShootDefaults.Pods
+				}
+			}
+		}
```

Comment:
something similar here and below if conditions also.
```suggestion
		if c.shoot.Spec.Networking.Pods == nil && !workerless &&
			c.seed.Spec.Networks.ShootDefaults != nil && c.seed.Spec.Networks.ShootDefaults.Pods != nil &&
			cidrMatchesIPFamily(*c.seed.Spec.Networks.ShootDefaults.Pods, c.shoot.Spec.Networking.IPFamilies) {
			c.shoot.Spec.Networking.Pods = c.seed.Spec.Networks.ShootDefaults.Pods
		}
```